### DISCUSSION TITLE
File upload save to custom relationship

### DISCUSSION THREAD

**Discussion by glenelkinsdev** [filamentphp/filament#11724] [Help]
### Package

Panel builder

### Package Version

3

### How can we help you?

I am using the file upload component in a form but i have a model File (files table) which i need to setup as a relationship to other model (the model the form is attached to)

There isn't any documentation showing that i can intercept the saving of the uploaded files once the form is submitted. I need to intercept as the files need to be stored in the files table with an entity relationship and not just saving the filename in the main model.

How can i do this?

**Comment by keasy9** [comment]
I had the same question and answer is to using `FileUpload::make('...')->saveUploadedFileUsing(function (BaseFileUpload $component, TemporaryUploadedFile $file, Model $record) {})`. But now I'm wondering if there's a way to show related file as uploaded in form?
